bin/od

```diff
@@ -68,7 +68,19 @@ $lastline = '';
 my $Program = basename($0);
 
 getopts('A:bcdfij:lN:ovx') or help();
-    defined $opt_A ? ($radix = $opt_A) : ($radix = 'o');
+if (defined $opt_A) {
+    if ($opt_A !~ m/^[doxn]$/) {
```

Comment:
I'd change that to /\A[doxn]\z/. The ^ and $ can allow more that what you are trying to validate, and change their effect with certain re module flags.